packages/flutter/test/material/app_bar_test.dart

```diff
@@ -2563,6 +2563,217 @@ void main() {
       expect(getAppBarBackgroundColor(tester), defaultColor);
       expect(tester.getSize(findAppBarMaterial()).height, kToolbarHeight);
     });
+
+    testWidgets('Material3 - scrolledUnderElevation should be maintained when drawer is opened', (WidgetTester tester) async {
```

Comment:
Can you add to this test to verify that after the drawer is closed the app bar responds to scrolling again as expected?